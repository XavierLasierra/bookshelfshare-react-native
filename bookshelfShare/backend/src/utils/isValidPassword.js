const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
