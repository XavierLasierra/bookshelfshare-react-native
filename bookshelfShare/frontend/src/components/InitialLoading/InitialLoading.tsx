import React, { useEffect } from 'react';
import {
  SafeAreaView,
  Text,
  TouchableOpacity
} from 'react-native';
import {
  SharedElement
} from 'react-navigation-shared-element';

import styles from './initialLoading.styles';

import BookIcon from '../../assets/bookIcon.svg';

export default function InitialLoading({
  navigation
}: any) {
  function handlePageChange() {
    navigation.push('Login');
  }

  useEffect(() => {
    setTimeout(handlePageChange, 2000);
  }, []);

  return (
    <SafeAreaView>
      <TouchableOpacity
        onPress={() => handlePageChange()}
        style={styles.titleContainer}
        activeOpacity={1}
      >
        <Text style={styles.title}>bookshelfShare</Text>
        <SharedElement id="mainIcon">
          <BookIcon
            width={50}
            height={50}
          />
        </SharedElement>
      </TouchableOpacity>
    </SafeAreaView>
  );
}
