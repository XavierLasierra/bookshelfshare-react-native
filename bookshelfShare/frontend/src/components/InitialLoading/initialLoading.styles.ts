import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  initialPage: {
    backgroundColor: stylesConstants.colors.main,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%'
  },
  titleContainer: {
    height: 30,
    alignItems: 'center'
  },
  title: {
    fontSize: 40,
    fontWeight: 'bold',
    color: stylesConstants.colors.mainText
  }
});
