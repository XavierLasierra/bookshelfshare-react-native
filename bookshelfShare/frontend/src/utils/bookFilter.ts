export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
