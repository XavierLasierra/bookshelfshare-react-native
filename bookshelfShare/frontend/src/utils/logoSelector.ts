import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
