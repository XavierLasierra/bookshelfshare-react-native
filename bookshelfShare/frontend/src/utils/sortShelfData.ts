export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
