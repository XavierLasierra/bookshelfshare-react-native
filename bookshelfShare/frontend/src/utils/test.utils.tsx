import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
