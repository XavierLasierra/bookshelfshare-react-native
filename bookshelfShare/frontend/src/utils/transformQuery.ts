interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
