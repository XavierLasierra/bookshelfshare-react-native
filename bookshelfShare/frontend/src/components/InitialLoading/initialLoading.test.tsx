import React from 'react';
import InitialLoading from './InitialLoading';
import { render } from '../../utils/test.utils';

import { automaticLogin } from '../../redux/actions/loggedUser.creator';
import loggedUserActions from '../../redux/actions/loggedUser.actions';

jest.mock('../../redux/actions/loggedUser.creator', () => ({
  automaticLogin: jest.fn()
}));

const navigation = {
  push: jest.fn()
};

jest.useFakeTimers();

describe('Given a InitialLoading component', () => {
  describe('When it is rendered', () => {
    test('Then should match the snapshot', () => {
      (automaticLogin as jest.Mock).mockReturnValue({
        type: ''
      });
      const screen = render(<InitialLoading navigation={navigation} />);
      expect(screen).toMatchSnapshot();
    });

    describe('And user needs login', () => {
      test('Then navigation.push should have been called with Login', () => {
        (automaticLogin as jest.Mock).mockReturnValue({
          type: loggedUserActions.USER_NOT_LOGGED
        });
        render(<InitialLoading navigation={navigation} />);
        jest.runOnlyPendingTimers();

        expect(navigation.push).toHaveBeenCalledWith('Login');
      });
    });

    describe('And user needs Login when rendering the screen', () => {
      test('Then automaticLogin should not be called', () => {
        (automaticLogin as jest.Mock).mockClear();
        const initialState = { loggedUser: { needsLogin: true } };
        render(<InitialLoading navigation={navigation} />, initialState);

        expect(automaticLogin).not.toHaveBeenCalled();
      });
    });
  });
});
