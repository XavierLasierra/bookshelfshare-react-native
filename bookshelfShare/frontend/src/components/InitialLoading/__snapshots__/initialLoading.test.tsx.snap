// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a InitialLoading component When it is rendered Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D9D8C4",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 30,
      }
    }
  >
    <View
      collapsable={false}
    >
      <IconMock
        height={60}
        width={60}
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
